examples_utils/benchmarks/run_benchmarks.py

```diff
@@ -152,6 +153,47 @@ def proc_thread():
     return (output, err, exitcode)
 
 
+def install_patched_requirements(requirements_file: Union[str, Path], listener: TextIOWrapper):
+    """Removes any 'examples-utils' requirements from a a requirements
+    file before installing it. It returns the original unpatched requirements
+    in case they are needed later."""
+
+    requirements_file = Path(requirements_file)
+    logger.info(f"Install python requirements")
+    if not requirements_file.exists():
+        raise FileNotFoundError(f"Invalid python requirements where specified at {requirements_file}")
+    # Strip examples-utils requirement as it can break the installation
+    original_requirements = requirements_file.read_text()
+    requirements_file.write_text("\n".join(l for l in original_requirements.splitlines() if "examples-utils" not in l))
+    cmd = [sys.executable, "-m", "pip", "install", "-r", str(requirements_file)]
+    out, err, exit_code = run_and_monitor_progress(cmd, listener)
+    if exit_code:
+        raise subprocess.CalledProcessError(exit_code, cmd, out, err)
+    return original_requirements
+
+
+def install_apt_packages(requirements_file_or_list: Union[str, Path, List[str]], listener: TextIOWrapper):
+    """Installs system packages with apt."""
+    logger.info(f"Installing apt requirements")
+    if not isinstance(requirements_file_or_list, list):
+        requirements_file = Path(requirements_file_or_list)
+        if not requirements_file.exists():
+            raise FileNotFoundError(f"Invalid apt requirements where specified at {requirements_file}")
+        requirements_list: List[str] = requirements_file.read_text().splitlines()
+    else:
+        requirements_list = requirements_file_or_list
+    logger.debug(f"  Collected the following requirements: " + " ".join(requirements_list))
+    for cmd in [
+        ["apt", "update", "-y"],
+        ["apt", "install", "-y", *requirements_list],
+    ]:
+        out, err, exit_code = run_and_monitor_progress(cmd, listener)
+        if exit_code:
+            raise subprocess.CalledProcessError(exit_code, cmd, out, err)
```

Comment:
Could we have this error output both to logs and stderr?

what we do currently is:
```
err = (f"Benchmark {benchmark_name} not found in any of the provided spec files, exiting.")
logger.error(err)
raise ValueError(err)
```

Every time we raise an error. To do with the philosophy of logging everything being output into one file (output.txt) (example stdout and stderr being seperate ofc)

Reply:
Yep, makes sense! Will add.